kernel/src/snapshot.rs

```diff
@@ -69,10 +69,7 @@ impl LogSegment {
         &self,
         engine_interface: &dyn EngineInterface,
     ) -> DeltaResult<Option<(Metadata, Protocol)>> {
-        let schema = Arc::new(StructType::new(vec![
-            crate::actions::schemas::METADATA_FIELD.clone(),
-            crate::actions::schemas::PROTOCOL_FIELD.clone(),
-        ]));
+        let schema = get_log_schema().project_as_schema(&[METADATA_NAME, PROTOCOL_NAME])?;
```

Comment:
rescuing https://github.com/delta-incubator/delta-kernel-rs/pull/129/files#r1527196393

> similar to other comment -- now this is only correct if metadata comes before protocol in the log schema.

I don't think that's true, we use `get_index_of` on the set of names, so it'll just find the correct items no matter the order given here. I've switched from `&[METADATA_NAME, PROTOCOL_NAME]` in the old PR to `&[PROTOCOL_NAME, METADATA_NAME]` to prove it works :)